model_api/cpp/models/src/segmentation_model.cpp

```diff
@@ -54,6 +68,34 @@ cv::Mat create_hard_prediction_from_soft_prediction(const cv::Mat& soft_predicti
     }
     return hard_prediction;
 }
+
+std::vector<Annotation> create_annotation_from_segmentation_map(const cv::Mat& hard_prediction, const cv::Mat& soft_prediction, std::vector<std::string> label_map) {
+    std::vector<Annotation> annotations = {};
+    cv::Mat label_index_map;
+    cv::Mat current_label_soft_prediction;
+    size_t labelID = 0;
+    for (int index = 1; index < soft_prediction.channels(); index++) {
+        cv::extractChannel(soft_prediction, current_label_soft_prediction, index);
+        cv::inRange(hard_prediction, cv::Scalar(index, index, index), cv::Scalar(index, index, index), label_index_map);
+        std::vector<std::vector<cv::Point>> contours;
+        cv::findContours(label_index_map, contours, cv::RETR_EXTERNAL, cv::CHAIN_APPROX_NONE);
```

Comment:
Why is the contour based on `hard_prediction`, but `probability` is based on `soft_prediction`, should they be the same?

Reply:
Was implemented this way by OTX. Seems that it can just use soft_prediction data. Will simplify thanks

Reply:
I think you still use both `hard_prediction` and `soft_prediction`